src/scripts/modules/list.js

```diff
@@ -52,50 +50,71 @@ function renderDestination(destination) {
   elDestination.classList.add("gallery-img--container");
 
   // add the correct class to elDestination wrapper
-  if (destination.visited === true) {
-    elDestination.classList.add("visited");
-  } else {
-    elDestination.classList.add("bucketlist");
-  }
+  destination.visited === true
+    ? elDestination.classList.add("visited")
+    : elDestination.classList.add("bucketlist");
 
   // create img element
   const img = document.createElement("img");
   img.src = destination.photo;
   img.classList.add("gallery-img");
+
   //create checkbox
   const checkbox = document.createElement("input");
   checkbox.type = "checkbox";
-  checkbox.classList.add("check");
-
-  // move checked items to 'visited' list
+  checkbox.classList.add("gallery-checkbox");
 
-  checkbox.addEventListener("click", () => {
-    toggleButtonVisited(destination);
-  });
-
-  //check all visited list checkboxes by default
+  //check all visited list checkboxes
   if (destination.visited) {
     checkbox.setAttribute("checked", "checked");
   }
 
-  // add img to destination
-  elDestination.appendChild(img);
-  //add checkbox
-  elDestination.appendChild(checkbox);
+  // create delete button
+  const deleteBtn = document.createElement("button");
+  deleteBtn.classList.add("gallery-delete-btn");
+  deleteBtn.innerText = "Delete"; //RESTORE DELETE BTN
 
-  //set img captions
+  //create img dropdown menu
+  const dropdown = document.createElement("div");
+  // dropdown.type = "button";
+  dropdown.classList.add("gallery-dropdown-icon");
+
+  //create figure caption
   const figcaption = document.createElement("figcaption");
   figcaption.classList.add("caption");
-  figcaption.insertAdjacentHTML(
+  //set captions in a div
+  const captionTxt = document.createElement("div");
+  captionTxt.classList.add("caption-text");
+  figcaption.appendChild(checkbox);
+  figcaption.appendChild(captionTxt);
+
+  captionTxt.insertAdjacentHTML(
     "beforeend",
     `<h2 class='caption__city'>${destination.city}</h2>`
   );
-  figcaption.insertAdjacentHTML(
+  captionTxt.insertAdjacentHTML(
     "beforeend",
     `<h3 class='caption__country'>${destination.country}</h3>`
   );
-  // add captions to destination
+
+  //Events
+  // move checked items to 'visited' list
+  checkbox.addEventListener("click", () => {
+    toggleButtonVisited(destination);
+    setTimeout(render, 1000);
```

Comment:
Do you mind explaining this timeout? What lead you to do that?

Reply:
I thought toggling was too sudden and I wanted feedback before the item is gone, i.e. to see the checkbox change or maybe do a cross out animation. Also adding and deleting take a bit so I also though this evens all three out

Reply:
Oh, I see!

There's a bug here. In slow internets (my current net is slow) this will break. the render might get called before the database process is finished.
My suggestion: Remove this render in this file and move it to `store.js`, to be called after the Parse method. That way we are sure that it gets called only after Parse is successful!

<details>
<summary>Check solution after you try it by yourself:</summary>

```js
  try {
    const object = await updateQuery[0].set("visited", item.visited).save();
    console.log("The object was updated successfully.");
     setTimeout(retreiveList, 1000);
  } catch (error) {
    console.log(error);
  }
```

</details>